File: backend/src/main/java/com/plantcare/plantcareassistant/entities/UserPlant.java
package com.plantcare.plantcareassistant.entities;

import jakarta.persistence.*;

import lombok.*;

import java.time.LocalDateTime;
import java.time.LocalTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@Entity
@Table(name = "user_plants")
public class UserPlant {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String apiPlantId;
    private String scientificName;
    private String customName;
    private String pictureUrl;

    private Boolean notificationsEnabled;
    private LocalTime notificationTime;
    private int wateringFrequency; // Frequency in days
    private String frequencyUnit; // "days", "weeks", "months"
}

File: AndroidApp/app/src/main/java/com/example/plantcare/data/model/UserPlant.kt
package com.example.plantcare.data.model

import java.time.LocalTime

data class UserPlant(
    val id: Long,
    val apiPlantId: String,
    val scientificName: String?,
    val customName: String?,
    val pictureUrl: String?,
    val notificationsEnabled: Boolean?,
    val notificationTime: LocalTime?,
    val wateringFrequency: Int, // Frequency in days
    val frequencyUnit: String // "days", "weeks", "months"
)

